packages/lambdas/src/shared/converter.ts

```diff
@@ -0,0 +1,211 @@
+import { DynamoDB } from "aws-sdk";
+import axios from "axios";
+import dayjs from "dayjs";
+import { capture } from "./capture";
+import { Log } from "./log";
+import { sleep } from "./sleep";
+
+// Keep this as early on the file as possible
+capture.init();
+
+// const converterKeysTableName = getEnvOrFail("FHIR_T0_CDA_CONVERTER_KEYS_TABLE_NAME");
+enum converterKeysStatus {
+  active = "active",
+  rateLimit = "rate-limit",
+  revoked = "revoked",
+}
+const MAX_CONVERTER_ATTEMPTS = 5;
+const CONVERTER_INITIAL_TIME_BETTWEEN_ATTEMPTS_MILLIS = 500;
+
+const createFhirConverter = (axiosTimeoutSeconds: number) =>
+  axios.create({
+    // Only response timeout, no option for connection timeout: https://github.com/axios/axios/issues/4835
+    timeout: axiosTimeoutSeconds * 1_000, // should be less than the lambda timeout
+    transitional: {
+      // enables ETIMEDOUT instead of ECONNABORTED for timeouts - https://betterstack.com/community/guides/scaling-nodejs/nodejs-errors/
+      clarifyTimeoutError: true,
+    },
+  });
+
+export async function postToConverter({
+  url,
+  payload,
+  axiosTimeoutSeconds,
+  converterKeysTableName,
+  log,
+  contentType,
+}: {
+  url: string;
+  payload: unknown;
+  axiosTimeoutSeconds: number;
+  converterKeysTableName: string;
+  log: Log;
+  contentType?: string;
+}) {
+  const sidechainUrl = url;
+  let attempt = 0;
+  let timeBetweenAttemptsMillis = CONVERTER_INITIAL_TIME_BETTWEEN_ATTEMPTS_MILLIS;
+  let apiKey: string;
+  while (attempt++ < MAX_CONVERTER_ATTEMPTS) {
+    apiKey = await getAndUpdateSidechainConverterKeys(converterKeysTableName);
+    log(`(${attempt}) Calling sidechain converter on url ${sidechainUrl}`);
+    try {
+      const fhirConverter = createFhirConverter(axiosTimeoutSeconds);
+
+      const res = await fhirConverter.post(sidechainUrl, payload, {
```

Comment:
Just a reminder, [this PR](https://github.com/metriport/metriport/pull/992) changed the logic to interact with the sidechain, let's make sure we're syncing so we don't lose the improvements done there.